reframe/frontend/loader.py

```diff
@@ -129,16 +135,24 @@ def load_from_module(self, module):
                 f'in test files: please use @reframe.simple_test decorator'
             )
 
-        if not hasattr(module, '_rfm_gettests'):
+        # FIXME: Remove the legacy_registry after dropping parameterized_test
+        registry = getattr(module, '_rfm_test_registry', None)
+        legacy_registry = getattr(module, '_rfm_gettests', None)
+        if not any((registry, legacy_registry)):
             getlogger().debug('No tests registered')
             return []
 
-        candidates = module._rfm_gettests()
-        if not isinstance(candidates, collections.abc.Sequence):
-            getlogger().warning(
-                f'Tests not registered correctly in {module.__name__!r}'
-            )
-            return []
+        candidates = registry.instantiate_all() if registry else []
+        legacy_candidates = legacy_registry() if legacy_registry else []
+        for c in (candidates, legacy_candidates):
+            if not isinstance(c, collections.abc.Sequence):
+                getlogger().warning(
+                    f'Tests not registered correctly in {module.__name__!r}'
+                )
+                return []
```

Comment:
To be honest, I think that this is a leftover from the days that user had to provide the `_get_checks()` method that instantiated the tests, so we had to check that what they returned was a sequence. We can remove this completely.

Reply:
Yeah, I also noticed that this was not covered by the tests.